invenio_oauthclient/contrib/keycloak/helpers.py

```diff
@@ -53,7 +53,45 @@ def get_public_key(remote, realm_url):
     return certs_resp["public_key"]
 
 
-def get_user_info(remote, resp_token, fallback_to_endpoint=True, options=dict()):
+def _get_user_info_from_token(remote, token, config_prefix):
+    """Get the user information from the JWT token."""
+    # try to parse the "id_token" part of Keycloak's (JWT) response
+    realm_url = current_app.config[f"{config_prefix}_REALM_URL"]
+    pubkey = _format_public_key(get_public_key(remote, realm_url))
+    alg = jwt.get_unverified_header(token)["alg"]
+
+    if not isinstance(options, dict):
+        options = {}
+
+    # consult the config whether to check the target audience
+    should_verify_aud = current_app.config.get(f"{config_prefix}_VERIFY_AUD", False)
+    expected_aud = current_app.config.get(f"{config_prefix}_AUD", None)
+
+    if should_verify_aud and (expected_aud is not None):
+        options.update({"verify_aud": True})
+    else:
+        options.update({"verify_aud": False})
+
+    # consult the config whether to check signature expiration
+    should_verify_expiration = current_app.config.get(
+        f"{config_prefix}_VERIFY_EXP", False
+    )
+
+    if should_verify_expiration:
+        options.update({"verify_exp": True})
+    else:
+        options.update({"verify_exp": False})
+
+    return jwt.decode(
+        token, key=pubkey, algorithms=[alg], audience=expected_aud, options=options
+    )
+
+def _get_user_info_from_endpoint(remote, config_prefix):
+    """Get the user info from the oauth server provider."""
+    url = current_app.config[f"{config_prefix}_USER_INFO_URL"]
+    return remote.get(url).data
+
+def get_user_info(remote, resp_token):
```

Comment:
minor: the docstring should be updated accordingly

(looks like there never was a `resp` argument... i think it references `resp_token` :sweat_smile: )